pubsub/iterator.go

```diff
@@ -624,3 +723,66 @@ func maxDuration(x, y time.Duration) time.Duration {
 	}
 	return y
 }
+
+const (
+	transientErrStringPrefix     = "TRANSIENT_"
+	transientInvalidAckErrString = transientErrStringPrefix + "FAILURE_INVALID_ACK_ID"
```

Comment:
This should never happen. Invalid ack id is a permanent failure, always. I do not see this variable used anywhere apart from tests so maybe the logic is right, just that we don't need this variable.

Reply:
Ah, I wasn't aware of this. I was mostly copying [Python's error strings](https://github.com/googleapis/python-pubsub/blob/e600ad8228930445765ffa0c45500a7779e25817/tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py#L2187) though I removed the variable anyway and made it a generic "TRANSIENT_FAILURE" error string for tests.